libass/ass.c

```diff
@@ -590,15 +597,68 @@ static int process_style(ASS_Track *track, char *str)
 
 }
 
+// WILL change contents of fmt1 and fmt2
+static int format_line_compare(char *fmt1, char *fmt2)
```

Comment:
Destructive comparison functions are evil. Maybe something like that would be better:
```c
// pattern is a list of words separated by ", "
bool format_line_compare(const char *fmt, const char *pattern)
{
    skip_spaces(&fmt);
    while (true) {
        if (*pattern == '\0') {
            skip_spaces(&fmt);
            return *fmt == '\0';
        }
        if (*pattern == ',') {
            skip_spaces(&fmt);
            if (*fmt++ != ',')
                return false;
            skip_spaces(&fmt);
            pattern += 2;  // skip space after comma
            continue;
        }
        if (lowertab[(uint8_t) *fmt++] != lowertab[(uint8_t) *pattern++])
            return false;
    }
}
```

Reply:
I didn't like this either, but was reluctant to duplicate the token.
I now put most of the non-destructive token-logic into the `inline` function `advance_token_pos`.
This should not affect the performance of `next_token` negatively.
 I should point out though, that with this change `next_token` does no longer replace **all** `','` with `'\0'`, but only if a token is terminated by a `','`. As far as I can tell, replacing them all was superfluous anyway.

Unfortunately `advance_token_pos` can't just have `const char**` arguments, as `(r)skip_spaces` takes `char**`, even though they don't modify the `char`-data. The compiler doesn't like passing `char**` to `const char**`.
For this reason `strdup` is still used, the alternative would be casting between `const char**` and `char**` at the necessary places.

Reply:
It's perfectly normal to cast to `(char *)` in such cases.

Reply:
Done.